modules/gapi/test/common/gapi_tests_common.hpp

```diff
@@ -469,6 +579,9 @@ template<typename Elem>
 using compare_vector_f = std::function<bool(const std::vector<Elem> &a,
                                             const std::vector<Elem> &b)>;
 
+template<typename Elem, int cn>
+using compare_vec_f = std::function<bool(const cv::Vec<Elem, cn> &a, const cv::Vec<Elem, cn> &b)>;
```

Comment:
Could you try `cv::norm` as comparator? If it works for every case I would remove the custom comparator

Reply:
Rewrote comparator using `cv::norm`. However, I chose to preserve comparator as I used some checks to avoid division by zero and I would like to cover it into function/comparator